### ISSUE TITLE
Implement subscriptions for paypal

### ISSUE THREAD

**Issue by clarete** [opencollective/opencollective#1141]
After we released PayPal for one time donations to collectives under the Open Source Collective, we need to implement PayPal in our subscription system.

The one time donations were implemented using the `v1/payments` path of the PayPal API. And unfortunately that isn't really possible to store an authorization token in a vault like we could in the Braintree API. With that in mind, we might need to implement their entire subscription flow instead of using our `charge_subscriptions` script.

Also, here's a note about PayPal requiring credit cards for subscriptions that might make our lives a lil harder: 

https://patreon.zendesk.com/hc/en-us/articles/115001917686-Why-do-I-need-to-add-a-credit-card-for-PayPal-

**Comment by mojavelinux** [comment]
You mention that it's possible for people to donate via PayPal. How does this work exactly? Is there something I need to do to enable it?

**Comment by clarete** [comment]
Hi @mojavelinux! Right now it's available for one time payments for all collectives hosted by the Open Source Collective. If you click the link "Or make a one time donation" underneath all the subscription tiers, you'll be directed to the `/donate` page of the collective. In that page, you need to choose "Paypal" under "Payment Methods" and then the paypal button will show up.

**Comment by mojavelinux** [comment]
Thank you! That worked!

**Comment by alannallama** [comment]
We still need this. It's a regularly requested feature when we ask for user feedback.

**Comment by josegonzalez** [comment]
Hi! It's not clear to me how much work this would be, but the Dokku collective would be willing to sponsor/co-sponsor a bounty for this issue. Feel free to reach out to me via email/slack (I'll re-add the OC slack tomorrow).

**Comment by alanna** [comment]
People still request this constantly. Would it be possible to at least scope it out so we can see how big a job it would be? @kewitz do you know how recurring payments work on paypal?

**Comment by nickcolley** [comment]
I was looking to do this today :+1:  My card declined as it expired and I feel safer with PayPal. :)

**Comment by TLWebdesign** [comment]
It's so time to get this going. We need this feature!!!

**Comment by Betree** [comment]
@TLWebdesign This is actually coming very soon, initial specifications are planned for this week and implementation should follow. If everything goes as expected, you can expect a release in less than a month.

**Comment by TLWebdesign** [comment]
Wow thats wonderful news :D THanks!

**Comment by Betree** [comment]
@TLWebdesign My initial estimation of 1 month was too optimistic, I'm sorry for sharing this misleading date. 

Our constraints (more especially, the fact that we need to support multiple fiscal hosts in different countries and with different accesses to PayPal's APIs) are making this more complex than expected and we are still researching and experimenting to make sure we implement this in the best way possible.

**To everyone following this issue, I can however ensure that [the project](https://github.com/opencollective/opencollective/issues/3986) is now officially kickstarted and will be a priority for the upcoming weeks.** I will share updates along the way to keep you up to date with our progress.

**Comment by Betree** [comment]
Project update: https://github.com/opencollective/opencollective/issues/3986#issuecomment-809282461

TLDR ; project was delayed because the initially chosen approach (Braintree) ended up being too complex for hosts to setup live, but we now have a working MVP relying on PayPal's Subscription API ready to be pushed to the next step. There are still some important paths to update (like the one to edit/cancel your recurring contribution) but the public release is getting closer!

**Comment by Betree** [comment]
We're currently doing some final tests internally, the feature should be released publicly next week. There's nothing to do on the host side if they already have a PayPal account connected, they will be migrated automatically.

**Comment by Betree** [comment]
The feature is now released and enabled for all hosts with a PayPal account. See https://github.com/opencollective/opencollective/issues/3986#issuecomment-828308074 for a full project update.

**Comment by smileBeda** [comment]
What are the ways to see this?
Clearly, in our OC instance, we cannot collect with PayPal, The only offered tools are bank account and stripe.
Both are very destructive - bank account is something no one will use to make a small donation, and stripe is _extremely selective, if not even discriminating_. The Stripe system will only accept you as a citizen of certain countries and only if you have a company UID _or_ a TAX ID in one of those countries.
To be precise, stripe is available in 47 countries out of ca. 200 (https://stripe.com/global).
Paypal is available in 200 countries and regions, see https://www.paypal.com/vn/webapps/mpp/country-worldwide

It is of crucial importance to anyone outside the "privilege zone" to have access to Paypal as a donation option. As this ticket states it was implemented, do we miss something in our OC instance, as we do not see it as an option?

Thanks.

**Comment by alanna** [comment]
@smileBeda we can usually turn on Paypal for any Host that requests it. Which Host are you with? @Betree can you turn it on for them?

**Comment by smileBeda** [comment]
Hi @alanna - thanks for your prompt feedback
Indeed I already communicated with support, and they did enable PayPal... for payouts! That is not what is meant to be implemented by this very GitHub task, as far I understand. Please correct me if I am wrong!

There is little point in having PayPal _payouts_ if all you can do is receive donations with a Bank Account 😓 
That only starts making sense once you can _receive_ donations with PayPal too

I also scanned the documentation, which is lacking PayPal in the section of "receive money", but mentions it in the "organisation settings", however doesn't mention it to be a _donation possibility_

Thus... I am (and others too) very confused as of how we can activate PayPal to receive donations. As said, Stripe is a very selective, dare I say almost discriminating-by-origin tool. Only 43 countries have access to it, and only if they are basically companies.

Thank you for understanding, and perhaps making it more clear on the documentations (or even the admin dashboards) about what a community can do in order to activate this!

**Comment by Betree** [comment]
Hi @smileBeda, sorry that was a mistake from our support. PayPal should now be enabled correctly on your account, you can follow this documentation for the next steps: https://docs.opencollective.com/help/fiscal-hosts/fiscal-host-settings/connected-accounts#receiving-contributions-through-paypal.

Let's continue in the email thread if you need assistance or have any other questions.
